metrics/release_test.go

```diff
@@ -17,239 +17,225 @@ limitations under the License.
-					Name: "release_attempt_concurrent_requests",
-					Help: "Total number of concurrent release attempts",
-				},
-			)
-			metrics.Registry.MustRegister(ReleaseAttemptRunningSeconds, ReleaseConcurrentTotal)
+			completionTime = &metav1.Time{}
+			startTime = &metav1.Time{Time: completionTime.Add(-60 * time.Second)}
 		})
 
-		AfterAll(func() {
-			metrics.Registry.Unregister(ReleaseAttemptRunningSeconds)
-			metrics.Registry.Unregister(ReleaseConcurrentTotal)
+		It("decrements ReleaseConcurrentTotal", func() {
+			Expect(testutil.ToFloat64(ReleaseConcurrentTotal.WithLabelValues())).To(Equal(float64(0)))
+			RegisterCompletedRelease(startTime, completionTime, "", "", "", "", "", "", "")
+			Expect(testutil.ToFloat64(ReleaseConcurrentTotal.WithLabelValues())).To(Equal(float64(-1)))
 		})
 
-		// Input seconds for duration of operations less or equal to the following buckets of 1, 5, 10 and 30 seconds
-		inputSeconds := []float64{1, 3, 8, 15}
-		elapsedSeconds := 0.0
+		It("adds an observation to ReleaseDurationSeconds", func() {
+			RegisterCompletedRelease(startTime, completionTime,
+				releaseDurationSecondsLabels[0],
+				releaseDurationSecondsLabels[1],
+				releaseDurationSecondsLabels[2],
+				releaseDurationSecondsLabels[3],
+				releaseDurationSecondsLabels[4],
+				releaseDurationSecondsLabels[5],
+				releaseDurationSecondsLabels[6],
+			)
+			Expect(testutil.CollectAndCompare(ReleaseDurationSeconds,
+				newHistogramReader(
+					releaseDurationSecondsOpts,
+					releaseDurationSecondsLabels,
+					startTime, completionTime,
+				))).To(Succeed())
+		})
 
-		It("increments the 'release_attempt_concurrent_total'.", func() {
-			creationTime := metav1.Time{}
-			for _, seconds := range inputSeconds {
-				startTime := metav1.NewTime(creationTime.Add(time.Second * time.Duration(seconds)))
-				elapsedSeconds += seconds
-				RegisterNewRelease(creationTime, &startTime)
-			}
-			Expect(testutil.ToFloat64(ReleaseConcurrentTotal)).To(Equal(float64(len(inputSeconds))))
+		It("increments ReleaseTotal", func() {
+			Expect(testutil.ToFloat64(ReleaseTotal.WithLabelValues("", "", "", "", "", "", ""))).To(Equal(float64(0)))
+			RegisterCompletedRelease(startTime, completionTime, "", "", "", "", "", "", "")
+			Expect(testutil.ToFloat64(ReleaseTotal.WithLabelValues("", "", "", "", "", "", ""))).To(Equal(float64(1)))
 		})
 
-		It("registers a new observation for 'release_attempt_running_seconds' with the elapsed time from the moment"+
-			"the Release was created to when it started (Release marked as 'Running').", func() {
-			// Defined buckets for ReleaseAttemptRunningSeconds
-			timeBuckets := []string{"1", "5", "10", "30"}
-			data := []int{1, 2, 3, 4}
-			readerData := createHistogramReader(AttemptRunningSecondsHeader, timeBuckets, data, "", elapsedSeconds, len(inputSeconds))
-			Expect(testutil.CollectAndCompare(ReleaseAttemptRunningSeconds, strings.NewReader(readerData))).To(Succeed())
+		It("increments ReleaseTotal", func() {
```

Comment:
same name as line 66

Reply:
@davidmogar can you use a different It string? You have `increments ReleaseTotal` twice in a row

Reply:
Actually is the same test. I removed the one that doesn't use labels.